### DISCUSSION TITLE
External auth, require re-authentication

### DISCUSSION THREAD

**Discussion by nick-merrill** [bluenviron/mediamtx#1053] [Questions] [answered]
Hello! Does anyone know whether there's a way to set a re-authentication period for external auth? An alternative might be to set a maximum connection duration, forcing a client to reconnect and to re-authenticate.

The problem I'm seeing is that once a client connects (and authenticates), they are not required to re-authenticate again until their connection is lost, even if their user is deactivated or password changed.

Open to alternatives to accomplish the same thing. One idea I had was to remove all paths that the user might be connect to and re-add them again, but that is very heavy-handed.

Thanks!

**Comment by aler9** [comment, marked as answer]
Hello, the problem of your approach is that if we require a periodic re-authentication, all clients would be periodically disconnected, and the received stream would be discontinuous, not an ideal solution for most users.

In my opinion you can use the API in order to manually kick all clients after a certain period. You don't need to remove all paths, you just have to kick users.

**Reply by nick-merrill** [reply]
Thank you! Manually kicking users when we need to sounds like a good approach.

[a comment by an automated account is not shown]
